src/box/lua/upgrade.lua

```diff
@@ -1365,6 +1365,74 @@ local function upgrade_to_2_11_1()
     drop_schema_max_id()
 end
 
+--------------------------------------------------------------------------------
+-- Tarantool 2.11.4
+--------------------------------------------------------------------------------
+local function create_persistent_gc_state_if_not_exists(issue_handler)
+    -- The function is also called on downgrade
+    if issue_handler ~= nil and issue_handler.dry_run then
+        return
+    end
+
+    local _space = box.space[box.schema.SPACE_ID]
+    local _index = box.space[box.schema.INDEX_ID]
+    local _priv = box.space[box.schema.PRIV_ID]
+    local _func = box.space[box.schema.FUNC_ID]
+    local space_id = box.schema.GC_CONSUMERS_ID
+    local opts = {group_id = 1}
+
+    -- Space already exists - do nothing
+    if _space:get(space_id) ~= nil then
+        return
+    end
+
+    -- We should create space and its index atomically so that we can
+    -- rely only on the fact that the space is created
+    log.info("open transaction to atomically create persistent WAL GC state")
+    box.begin()
+
+    log.info("create persistent trigger to persist gc consumers on replicas")
+    local trigger_body_raw = [[function(old, new)
```

Comment:
When upgrading schema in a cluster, `box.schema.upgrade()` is called only on master, and we need to create gc consumers on all instances strictly after space `_gc_consumers` is created, so that's why a trigger that will be replicated to all instances.

Dropped a comment here.

Reply:
Alternatively you could make this trigger part of `alter.cc` without all the hassle with creating a persistent trigger.